snapshot-labs/sx-monorepo
# feat: add EthRelayer execution support on L2

### Summary

This PR adds EthRelayer support. Only part missing is performing execution on L1 which depends on some contract-level changes that we need to wait for.

Depends on: https://github.com/snapshot-labs/sx-monorepo/pull/321

### How to test

1. Run local services (`yarn dev:full`).
2. Create space with EthRelayer execution strategy. Add some Safe address as avatar (it has to have some funds in it). Give it some short max voting delay.
3. Create proposal with execution.
4. Vote on it.
5. Wait for proposal to end.
6. Execute button should show up (it might fail for some time due to timestamps being not accurate initially, will need to look into it).
7. Click on it, execution on L2 should be triggered.

### To-Do

- [x] Add missing fields to Subgraph APIs.
- [x] Add `executors_destinations` to metadata as source for API.
- [x] Add Starknet execution through mana (currently done on UI).
- [ ] Finish execution on L1 (future PR).
